src/custom-element.js

```diff
@@ -380,7 +376,16 @@ function createBaseCustomElementClass(win) {
         // Already upgraded or in progress or failed.
         return;
       }
-      this.implementation_ = new newImplClass(this);
+
+      // If the implmementation fails to construct, we'll leave it as a
+      // ElementStub. The runtime will ignore the element from now on.
+      try {
+        this.implementation_ = new newImplClass(this);
+      } catch (e) {
```

Comment:
It's so that we don't continue to treat this element as if it were upgraded. It's a copy from https://github.com/ampproject/amphtml/pull/25382, where we had to wrap in a try-catch to make sure that we always had an implementation.

Reply:
There's already a try/catch around the caller of this method -- we could just update that error report to contain the element tag.

Reply:
There are two callers:

- `connectedCallback`
- `upgradeOrRegisterElement`

Only `upgradeOrRegisterElement` has a try-catch. I think this is the right spot for the catch.